mysql/Makefile

```diff
@@ -29,11 +32,19 @@ generate:
 	go get -d -t -v ./...
 
 reset-db: require-client
-	SQL="" && \
-	SQL+="DROP DATABASE IF EXISTS $(DB_NAME);" && \
-	SQL+="CREATE DATABASE $(DB_NAME);" && \
-	SQL+="GRANT ALL PRIVILEGES ON $(DB_NAME).* TO $(DB_USERNAME) IDENTIFIED BY '$(DB_PASSWORD)';" && \
-	mysql -uroot -h"$(DB_HOST)" -P$(DB_PORT) <<< $$SQL
+	@if [ -z "$(MYSQL_ROOT_PASSWORD)" ]; then \
+	       SQL="" && \
+	       SQL+="DROP DATABASE IF EXISTS $(DB_NAME);" && \
+	       SQL+="CREATE DATABASE $(DB_NAME);" && \
+	       SQL+="GRANT ALL PRIVILEGES ON $(DB_NAME).* TO $(DB_USERNAME) IDENTIFIED BY '$(DB_PASSWORD)';" && \
+	       mysql -u$(MYSQL_ROOT_USERNAME) -h$(DB_HOST) -P$(DB_PORT) <<< $$SQL; \
```

Comment:
This is the only line that changes, could we check for `MYSQL_ROOT_PASSWORD` here?